Sources/RemoteMessaging/Mappers/JsonToRemoteMessageModelMapper.swift

```diff
@@ -161,8 +169,23 @@ struct JsonToRemoteMessageModelMapper {
             return .share(value: jsonAction.value, title: jsonAction.additionalParameters?["title"])
         case .url:
             return .url(value: jsonAction.value)
-        case .surveyURL:
-            return .surveyURL(value: jsonAction.value)
+        case .survey:
+            if let queryParamsString = jsonAction.additionalParameters?["queryParams"] as? String {
+                let queryParams = queryParamsString.components(separatedBy: ";")
+                let mappedQueryParams = queryParams.compactMap { param in
+                    return RemoteMessagingSurveyActionParameter(rawValue: param)
+                }
+
+                if mappedQueryParams.count == queryParams.count, let surveyURL = URL(string: jsonAction.value) {
+                    let updatedURL = surveyActionMapper.add(parameters: mappedQueryParams, to: surveyURL)
+                    return .survey(value: updatedURL.absoluteString)
+                } else {
+                    // The message requires a parameter that isn't supported
+                    return nil
```

Comment:
@amddg44 The idea here is that if we add a new parameter and specify it in a survey, and version of the client that doesn't support the parameter will not display the message. Is this the correct way to handle that? In my testing it works as I want it to, but wanted to be sure.

Reply:
@samsymons yes, LGTM 👍